File: src/main/java/io/transwarp/logparser/filter/LevelFilter.java
package io.transwarp.logparser.filter;

import io.transwarp.logparser.util.Record;

import java.util.Optional;

/**
 * Author: stk
 * Date: 18/4/2
 */
public class LevelFilter implements Filter {
    @Override
    public boolean filter(Record record) {
        return Optional.of(record).map(Record::getLevel).filter(s -> s.equals("WARN") || s.equals("ERROR")).stream().count() == 1;
    }
}

File: src/main/scala/io/transwarp/logparser/filter/LevelFilter.scala
package io.transwarp.logparser.filter

import io.transwarp.logparser.util.LogEntry

/**
  * Author: stk
  * Date: 2018/4/4
  *
  * @param params levels
  */
class LevelFilter(override val params: List[String]) extends Filter {
  override def filter(logEntry: LogEntry): Boolean = logEntry.format("level").asInstanceOf[Option[String]]
    .fold(false)(params.contains(_))

  override def copy(): Filter = new LevelFilter(params)
}
